cache/src/main/java/com/kroger/cache/SnapshotFileCacheBuilder.kt

```diff
@@ -77,81 +72,41 @@ public interface SnapshotFileCacheBuilder<T> {
+         * to read and write to the cache [File].
          *
          * @param parentDirectory the directory to create and save the cache file in. On Android it is
          * recommended to be somewhere in the cache directory such as Context.getCacheDir.
          * @param filename this should be in the form of "com.example.module.feature".
          * It is important for the filename to be unique across all [Cache] instances.
-         * @param keySerializer serializer and deserializer for cache keys
-         * @param valueSerializer serializer and deserializer for cache values
-         * @param formatter [StringFormat] to use when doing [File] I/O.
-         * Defaults to [Json][kotlinx.serialization.json.Json].
-         */
-        public fun <K, V> from(
-            parentDirectory: File,
-            filename: String,
-            keySerializer: KSerializer<K>,
-            valueSerializer: KSerializer<V>,
-            formatter: StringFormat = Json,
-        ): SnapshotFileCacheBuilder<List<CacheEntry<K, V>>> {
-            val cacheEntrySerializer = ListSerializer(CacheEntry.serializer(keySerializer, valueSerializer))
-            return RealSnapshotFileCacheBuilder(
-                parentDirectory,
-                filename,
-                DefaultFileReader(cacheEntrySerializer, formatter),
-                DefaultFileWriter(cacheEntrySerializer, formatter),
-            )
-        }
-
-        /**
-         * This is a convenience function to create a [SnapshotFileCacheBuilder] that uses [StringFormat]
-         * from the kotlinx serialization library to read and write to the cache [File].
-         *
-         * @param parentDirectory the directory to create and save the cache file in. On Android it is
-         * recommended to be somewhere in the cache directory such as Context.getCacheDir.
-         * @param filename this should be in the form of "com.example.module.feature".
-         * It is important for the filename to be unique across all [Cache] instances.
-         * @param valueSerializer serializer and deserializer for cache values
-         * @param formatter [StringFormat] to use when doing [File] I/O.
-         * Defaults to [Json][kotlinx.serialization.json.Json].
+         * @param valueSerializer an implementation of [CacheSerializer]
          */
         public fun <T> from(
             parentDirectory: File,
             filename: String,
-            valueSerializer: KSerializer<T>,
-            formatter: StringFormat = Json,
+            valueSerializer: CacheSerializer<T>,
         ): SnapshotFileCacheBuilder<T> = from(
             parentDirectory,
             filename,
-            DefaultFileReader(valueSerializer, formatter),
-            DefaultFileWriter(valueSerializer, formatter),
+            DefaultFileReader(valueSerializer),
+            DefaultFileWriter(valueSerializer),
```

Comment:
Wondering if we shouldn't just make the file cache take a `CacheSerializer` at this point? Otherwise it should be something like this though and the default classes can be removed.

```suggestion
            valueSerializer::decodeFromString,
            valueSerializer::toByteArray,
```

Reply:
That's a good call out that I was too in the weeds to pick up on. Migrating `SnapshotFileCache` to take a `CacheSerializer`